packages/apps/slack/src/services/slack.ts

```diff
@@ -60,37 +61,31 @@ export default class SlackService extends TransactionBaseService {
 
     try {
       const slackChannels = await this.fetchSlackChannels(config)
-
       for (const channel of slackChannels) {
         const conversations = await this.fetchChannelConversations(channel.id,config);
         for(const conversation of conversations){
-          const doc : IndexableDocument = {
-            id: conversation.ts,
+          const thread = await this.fetchThreadForConversation(channel.id,conversation.ts,config)
+
+          const sections: Section[] = thread.map((message, index) => ({
+            offset: index,
+            content: message.text,
+            link: `https://slack.com/api/conversations.replies?channel_id=${channel.id}&ts=${conversation.ts}` 
+          }));
+          const threadDoc : IndexableDocument = {
+            id:conversation.ts, // conversation id
             organisationId:org.id,
-            title:conversation.text,
             source:AppNameDefinitions.SLACK,
-            updatedAt: new Date(Date.now()),
-            metadata:{sentByUser:conversation.user}
+            title:conversation.text, // the main message which lead to conversation
+            metadata:{channel_id:channel.id}, // passing channel id just for top down reference
+            sections:sections, // an array of messages in the specific conversation
+            updatedAt: new Date(Date.now())
```

Comment:
Add `type: DocType.TEXT` in Indexable Document

Reply:
in metadata?

Reply:
just like a title , add a field as `type : type: DocType.TEXT` , for reference you can checkout the jira app in Main ocular Repo